gst/nnstreamer/tensor_common.c

```diff
@@ -857,6 +857,65 @@ gst_tensors_config_from_structure (GstTensorsConfig * config,
   return TRUE;
 }
 
+/**
+ * @brief Get tensors caps from tensors config and caps of the src pad
+ * @param pad GstPad to check if it supports other/tensor
+ * @param config tensors config structure
+ * @return caps for given config and pad
+ */
+GstCaps *
+gst_tensors_get_caps (GstPad * pad, const GstTensorsConfig * config)
+{
+  GstCaps *caps = NULL;
+
+  if (config->info.num_tensors == 1 && gst_pad_has_tensor_caps (pad)) {
+    GstTensorConfig tensor_config;
+    tensor_config.info = config->info.info[0];
+    tensor_config.rate_n = config->rate_n;
+    tensor_config.rate_d = config->rate_d;
+    caps = gst_tensor_caps_from_config (&tensor_config);
+  } else {
+    caps = gst_tensors_caps_from_config (config);
```

Comment:
Can you guarantee that this is ok if pad already has (partially) configured GSTCAP? or has some ranges in its values? (types or dims)